drivers/adc/ad717x/ad717x.c

```diff
@@ -52,6 +52,333 @@
+ * @param analog_input - Analog Inputs to the Channel
+ * @return Returns SUCCESS for success or negative error code in case of failure.
+*****************************************************************************/
+int ad717x_connect_analog_input(ad717x_dev *device, uint8_t channel_id,
+				union ad717x_analog_inputs analog_input)
+{
+	ad717x_st_reg *channel_reg; // Pointer to the channel register
+
+	if (!device)
+		return INVALID_VAL;
+
+	/* Retrieve the channel register */
+	channel_reg = AD717X_GetReg(device, AD717X_CHMAP0_REG + channel_id);
+	if (!channel_reg)
+		return INVALID_VAL;
+
+	switch ((uint8_t)device->active_device) {
+	case ID_AD4111 :
+	case ID_AD4112 :
+	case ID_AD4114 :
+	case ID_AD4115 :
+		/* Clear and Set the required analog input pair to channel */
+		channel_reg->value  &= ~AD717x_CHANNEL_INPUT_MASK;
+		channel_reg->value |= AD4111_CHMAP_REG_INPUT(analog_input.analog_input_pairs);
+		if (AD717X_WriteRegister(device, AD717X_CHMAP0_REG + channel_id) < 0)
+			return INVALID_VAL;
+
+		device->chan_map[channel_id].analog_inputs.analog_input_pairs =
+			analog_input.analog_input_pairs;
+		break;
+
+	case ID_AD7172_4:
+	case ID_AD7173_8:
+	case ID_AD7175_2:
+	case ID_AD7175_8:
+	case ID_AD7176_2:
+	case ID_AD7177_2:
+	case ID_AD7172_2:
+		/* Select the Positive Analog Input */
+		channel_reg->value &= ~AD717X_CHMAP_REG_AINPOS_MSK;
+		channel_reg->value |=  AD717X_CHMAP_REG_AINPOS(
+					       analog_input.ainp.pos_analog_input);
+
+		/* Select the Negative Analog Input */
+		channel_reg->value &= AD717X_CHMAP_REG_AINNEG_MSK;
+		channel_reg->value |= AD717X_CHMAP_REG_AINNEG(
+					      analog_input.ainp.neg_analog_input);
+		if (AD717X_WriteRegister(device, AD717X_CHMAP0_REG + channel_id) < 0)
+			return INVALID_VAL;
+
+		device->chan_map[channel_id].analog_inputs.ainp.pos_analog_input =
+			analog_input.ainp.pos_analog_input;
+		device->chan_map[channel_id].analog_inputs.ainp.neg_analog_input =
+			analog_input.ainp.neg_analog_input;
+		break;
+
+	default :
+		return INVALID_VAL;
+	}
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Assign Setup to Channel
+ * @param device - AD717x Device Descriptor
+ * @param channel_id - Channel ID (number)
+ * @param setup - Setup ID (number)
+ * @return Returns SUCCESS for success or negative error code in case of failure.
+******************************************************************************/
+int ad717x_assign_setup(ad717x_dev *device, uint8_t channel_id, uint8_t setup)
+{
+	ad717x_st_reg *p_register;   // Pointer to the channel register
+
+	if (!device)
+		return INVALID_VAL;
+
+	/* Retrieve the Channel Register */
+	p_register = AD717X_GetReg(device, AD717X_CHMAP0_REG + channel_id);
+	if (!p_register)
+		return INVALID_VAL;
+
+	/* Assign set up to the chosen channel */
+	p_register->value =
+		(p_register->value & ~AD717X_CHMAP_REG_SETUP_SEL_MSK) |
+		AD717X_CHMAP_REG_SETUP_SEL(setup);
```

Comment:
nitpick: since until now we've done this in two steps (1. clear bit field; 2. write value;) let's try to be consistent. Either we change this to be in two steps, or we change the others to be in one step. I personally prefer the first, but it's up to you.

Reply:
Added two separate steps for clear and write value